### PR TITLE
[BE] Reformat ReduceOps

### PR DESCRIPTION
**Pull request by malfet**
Use curly braces even after single line `if`
Use whitespace between `if` and condition
Use `c10::irange`
Also, use `c10::multiply_integers` instead of explicit for loop of elements of `IntArrayRef`
Do not pass `num_input_dims` to `set_apparent_shapes` as it is always equal to the length of `input_shape` array

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
Why `c10::irange` is better than C-style loop?

**Review by huydhn — APPROVED** [review]
LGTM! Only minor linter issues

**Comment by malfet** [comment]
> Why `c10::irange` is better than C-style loop?

- It's automatically avoid `signed-unsigned comparison warning` by returning iterator of the same signedness as upper bound
- Declaring iterator const makes it explicit that it did not get altered in the loop body

File: aten/src/ATen/native/mps/operations/ReduceOps.mm [hunk 1]
```diff
@@ -132,30 +123,29 @@ void set_axes_and_shapes(const Tensor& input_t,
 
   // Squeeze dims for output shape
   output_shape = [NSMutableArray<NSNumber*> arrayWithCapacity:0];
-  for(int i=0; i < num_output_dims; i++) {
-    if([apparent_output_shape[i] longValue] != 1) {
+  for (auto i : c10::irange(num_output_dims)) {
```
Comment (kit1980) [inline aten/src/ATen/native/mps/operations/ReduceOps.mm:126]: `const auto i`?

**Pull request closed without merging**